sbin/prepareWorkspace.sh

```diff
@@ -286,8 +286,33 @@ updateOpenj9Sources() {
   # Building OpenJDK with OpenJ9 must run get_source.sh to clone openj9 and openj9-omr repositories
   if [ "${BUILD_CONFIG[BUILD_VARIANT]}" == "${BUILD_VARIANT_OPENJ9}" ]; then
     cd "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}" || return
+
+    if [ "$JAVA_FEATURE_VERSION" -eq 17 ]
+    then
+      if [[ $BUILD_ARGS == *"--bundle-openjceplus"* ]]; then
+        # Set the flags to get the OpenJCEPlus source code
+        OPENJCEPLUS_FLAGS="-openjceplus-repo=https://github.com/ibmruntimes/OpenJCEPlus.git -openjceplus-branch=semeru-main"
+        # Set the flags to get the appropriate GSKit binaries
+        GSKIT_FOLDER="https://na.artifactory.swg-devops.com/artifactory/sec-gskit-javasec-generic-local/gskit8/20230802_8.9.5"
+        if [ "$TARGET_OS" = "linux"  ]; then
+          if [ "$ARCHITECTURE" = "x64"  ]; then
+            GSKIT_FLAGS="-gskit-bin=${GSKIT_FOLDER}/linux64_x86/jgsk_crypto.tar -gskit-sdk-bin=${GSKIT_FOLDER}/linux64_x86/jgsk_crypto_sdk.tar"
+          elif [ "$ARCHITECTURE" = "ppc64le"  ]; then
+            GSKIT_FLAGS="-gskit-bin=${GSKIT_FOLDER}/linux64_ppcle/jgsk_crypto.tar -gskit-sdk-bin=${GSKIT_FOLDER}/linux64_ppcle/jgsk_crypto_sdk.tar"
+          fi
+        fi
+        if [ "$TARGET_OS" = "aix"  ]; then
+          GSKIT_FLAGS="-gskit-bin=${GSKIT_FOLDER}/aix64_ppc/jgsk_crypto.tar -gskit-sdk-bin=${GSKIT_FOLDER}/aix64_ppc/jgsk_crypto_sdk.tar"
+        fi
+        if [ "$TARGET_OS" = "windows"  ]; then
+          GSKIT_FLAGS="-gskit-bin=${GSKIT_FOLDER}/win64_x86/jgsk_crypto.tar -gskit-sdk-bin=${GSKIT_FOLDER}/win64_x86/jgsk_crypto_sdk.tar"
+        fi
```

Comment:
Are these artifactory links predictable? I'm wondering it we only need to have one `if` that says "is this a gskit platform?" and then set the link based on the OS/Arch values that should be available in the env vars already?
Might be easier said than done. `windows != win64`, `x86 != x64` etc. Not sure we want to bother with some custom mapping if we can just hardcode the 4-8 links like you have already done.

Reply:
I thought about it too, but, like you mention, there are a lot of differences for now. It's `linux64 != linux`, `ppcle != ppc` and others are going to come up as we add more platforms.

So, not sure if it's worth it.